### PR TITLE
Remove most outdated version notes

### PR DESCRIPTION
**Pull request by parlough**
With this change I kept the version notes about the old versions of the `dart` tools for now as their deprecation is pretty recent and I still see a fair bit of usage of the old commands in the wild. Do you think those should be removed now as well @kwalrath?

Fixes https://github.com/dart-lang/site-www/issues/2174

### CODE REVIEW

**Comment by kwalrath** [comment]
> With this change I kept the version notes about the old versions of the `dart` tools for now as their deprecation is pretty recent and I still see a fair bit of usage of the old commands in the wild. Do you think those should be removed now as well @kwalrath?

No, I think you're right to keep the tool notes in, for now.

File: src/_guides/language/extension-methods.md [hunk 1]
```diff
@@ -8,11 +8,6 @@ You might use extension methods without even knowing it.
 For example, when you use code completion in an IDE,
 it suggests extension methods alongside regular methods.
 
-{{site.alert.version-note}}
-  To use extension methods in a package,
-  the pubspec must have a [minimum SDK constraint][] of at least 2.7.0.
-{{site.alert.end}}
-
 [minimum SDK constraint]: /tools/pub/pubspec#sdk-constraints
```
Comment (kwalrath) [inline src/_guides/language/extension-methods.md:11]: Is this link used anywhere else? If not:

```suggestion
```

**Review by kwalrath — COMMENTED** [review]
Aside from a couple of small things, LGTM!

File: src/tools/index.md [hunk 2]
```diff
@@ -19,11 +19,10 @@ If you aren't sure which tools you need, **get the Flutter SDK.**
 
 [General-purpose tools]: #general-purpose-tools
 
-{{site.alert.version-note}}
-  As of Flutter 1.21, the Flutter SDK includes the full Dart SDK.
+{{site.alert.note}}
+The Flutter SDK includes the full Dart SDK.
```
Comment (kwalrath) [inline src/tools/index.md:23]: It's our standard to indent the note text 2 spaces. Idk why; I just obey.

```suggestion
  The Flutter SDK includes the full Dart SDK.
```

**Comment by parlough** [comment]
@kwalrath Thanks for the review. I've addressed your comments.

I've also created https://github.com/dart-lang/site-www/issues/3487 to track the future work on removing the old tool version notes.

File: src/tools/index.md (hunk 2 above)
Comment (parlough) [reply src/tools/index.md]: Oops, thanks! Lost them in the copy paste :)

**Review by kwalrath — APPROVED** [review]
lgtm. Thanks, @parlough!

**Pull request merged**